onnxscript/function_libs/torch_aten/param_manipulation.py

```diff
@@ -0,0 +1,171 @@
+"""Function for manipulating input parameters of an Op or a OnnxFunction."""
+from __future__ import annotations
+
+import collections
+import dataclasses
+from typing import Any, List, OrderedDict, Sequence, Tuple
+
+import onnx
+
+from onnxscript import values
+
+# A special value to indicate that the default value is not specified
+_EmptyDefault = object()
+
+
+@dataclasses.dataclass(frozen=True)
+class ParamSchema:
+    """A schema for a parameter of an Op or a OnnxFunction.
+
+    Attributes:
+        name: The name of the parameter.
+        type: The type of the parameter.
+        default: The default value of the parameter.
+        is_input: Whether the parameter is an ONNX input.
+    """
+
+    name: str
+    type: Any = None  # Op input does not have a type, for now
+    default: Any = _EmptyDefault
+    is_input: bool = True
+
+    def __repr__(self) -> str:
+        param_kind = "INPUT" if self.is_input else "ATTRIBUTE"
+        text = f"{self.name}<{param_kind}>: {self.type}"
```

Comment:
nit (optional): f"{self.name} : {param_kind}({self.type})" or f"{param_kind} {self.name} : {self.type}" would be more readable?

Reply:
SG!

Reply:
a<INTPUT>: INT64

vs

INPUT a : INT64
a : INTPUT(INT64)

Reply:
I will go with `a: INTPUT[INT64]`